### PR TITLE
refactor!: barrel export manually

### PR DESCRIPTION
**Pull request by SamuelAlev**
- Barrel export manually (folder renaming break the script), it makes things simplier to refactor.
- Added missing depedencies to `dependencies`, we were relying on subdependencies for 4 of them.
- Deleted `deepmerge` as it is no longer used.

### CODE REVIEW

**Review by jonasbuechel — CHANGES_REQUESTED** [review]
Hey Samuel, I had a look at PR613 and the changes are looking fine so far and Arcade works at deployed storybook.
I tried it in a fresh create-react app and bundle size drops from 1.190 MB to 0.891 MB :muscle: 

I’m having issues with testing it on the PR’s branch in clarify via `npm link` where it runs on Arcade’s main branch when using the same approach :thinking_face: 
Do you issue the same problems or is it maybe an issue by using npm link?

These are some examples, warnings turn into ERROR at the later output.

```
WARNING in ./application/client/Feature/Library/Collection/CollectionActionMenu/CollectionActionMenu.tsx 147:16-35
export 'ButtonRounding' (imported as 'ButtonRounding') was not found 
 WARNING in ./application/client/Feature/Library/Collection/CollectionActionMenu/CollectionActionMenu.tsx 163:34-44
[0] export 'ActionMenu' (imported as 'ActionMenu') was not found in '@frontify/arcade'


 (possible exports: Accordion, AccordionItem, AddBlockButton, AddBlockButtonDirection, AssetInput, AssetInputSize, Badge, BadgeStatus, BadgeStyle, BrandColorPicker, BreadcrumbItem, Breadcrumbs, BrightHeader, BrightHeaderStyle, Button, ButtonGroup, ButtonSize, ButtonStyle, ButtonType, CUSTOM_LINK_ID, Card, Checkbox, CheckboxState, Checklist, ChecklistDirection, Code, ColorFormat, ColorInput, ColorInputTitle, ColorInputTrigger, ColorPicker, ColorPickerFlyout, ColorPreview, CurrentBreadcrumbItem, CustomColorPicker, DEFAULT_ICON, Divider, DividerHeight, DividerStyle, Dropdown, DropdownSize, FLYOUT_DIVIDER_COLOR, FLYOUT_DIVIDER_HEIGHT, FOCUS_STYLE, FOCUS_STYLE_INSET, FieldsetHeader, FieldsetHeaderSize, FieldsetHeaderType, Flyout, FlyoutFooter, FocusController, FocusControllerWidth, FocusControllerWidthClass, FormControl, FormControlDirection, FormControlStyle, Heading, HelperPosition, Icon, IconAcademy, IconActions, IconActivities, IconActivity, IconAdd, IconAddSimple, IconAddToCollection, IconAdobeCreativeCloud, IconAlert, IconAlignBottom, IconAlignCenter, IconAlignLeft, IconAlignMiddle, IconAlignRight, IconAlignTop, IconAnalytics, IconAngleDown, IconAnimalsNature, IconAnnotations, IconAppearance, IconApprove, IconArrow, IconArrowDown, IconArrowLeft, IconArrowRight, IconArrowUp, IconArrowUpAndDown, IconAssets, IconAttachment, IconAttentionFilled, IconAttributes, IconAudio, IconBackward5Seconds, IconBold, IconBookmark, IconBrand, IconBriefing, IconBuilder, IconButton, IconCalendar, IconCallout, IconCampaign, IconCampaignActivity, IconCards, IconCaretDown, IconCaretLeft, IconCaretRight, IconCaretUp, IconCenter, IconCheck, IconCheckFilled, IconCircle, IconCollapse, IconCollectionLarge, IconCollectionMedium, IconCollectionSmall, IconColorScale, IconColorScaleSmall, IconColorScaleTall, IconColorSwatch, IconColors, IconColumnBreak, IconCopyToClipboard, IconCrop, IconCursorClick, IconCut, IconDivider, IconDoDontsStrikethrough, IconDoDontsUnderline, IconDocument, IconDocumentLibrary, IconDosImages, IconDosText, IconDoubleChevronLeft, IconDoubleChevronRight, IconDoubleQuotesDown, IconDoubleQuotesUp, IconDownload, IconDownloadAlternative, IconDrops, IconDuplicate, IconEmojiHappy, IconEmojiHappyFilled, IconEmojiNeutral, IconEmojiNeutralFilled, IconEmojiUnhappy, IconEmojiUnhappyFilled, IconEnum, IconExpand, IconExternalAsset, IconExternalLink, IconFeedback, IconFigureTextBottom, IconFigureTextLeft, IconFigureTextRight, IconFigureTextTop, IconFile, IconFiletype, IconFilter, IconFitToScreen, IconFlags, IconFocalPoint, IconFolder, IconFolderUp, IconFont, IconFontKit, IconFoodDrink, IconForward5Seconds, IconFrequentlyUsed, IconFullwidth, IconGeneral, IconGenericPost, IconGuidelines, IconHeader, IconHighlight, IconHookBracketLeft, IconHookBracketRight, IconIconAndText, IconIcons, IconIframe, IconImage, IconImageAndText, IconImageFigure, IconImageGrid2, IconImageGrid3, IconImageGrid4, IconImageLibrary, IconImageTextRatio25, IconImageTextRatio33, IconImageTextRatio50, IconImageTextRatio66, IconImageTextRatio75, IconIndesign, IconInfo, IconIntegration, IconIntegrations, IconInteractions, IconItalic, IconLabel, IconLayoutGrid, IconLayoutHorizontal, IconLayoutVertical, IconLightning, IconLightningFilled, IconLineDashes, IconLineDotted, IconLineSolid, IconLineSpacer, IconLink, IconListBullets, IconListChecklist, IconListIndented, IconListNumbers, IconLock, IconLogo, IconLogoGrid, IconLogout, IconMarkArea, IconMeasurements, IconMedia, IconMegamenu, IconMetadata, IconMinus, IconModules, IconMore, IconMove, IconMovePage, IconNone, IconNotifications, IconObjects, IconOpenLock, IconOpenLockFilled, IconOptions, IconOrientation, IconPatternLibrary, IconPause, IconPen, IconPills, IconPin, IconPinFilled, IconPlanning, IconPlay, IconPlus, IconPointOut, IconPolygon, IconProcessing, IconProjects, IconPublication, IconQuestion, IconQuote, IconRectangle, IconRefresh, IconReject, IconRejectCircle, IconRejectFilled, IconRevert, IconRevisions, IconRocket, IconSearch, IconSection, IconSettings, IconShare, IconSide, IconSingleChevronLeft, IconSingleChevronRight, IconSingleQuoteDown, IconSingleQuoteUp, IconSize, IconSizeMap, IconSketch, IconSmileysPeople, IconSnippet, IconSpinner, IconStar, IconStarFilled, IconStorybook, IconStrikethrough, IconSubmit, IconSymbols, IconTable, IconTag, IconTarget, IconTeam, IconTeamO, IconTemplate, IconTemplates, IconText, IconTextAlignCenter, IconTextAlignJustify, IconTextAlignLeft, IconTextAlignRight, IconTextLibrary, IconTextMultiColumn, IconTextOnImage, IconTextSnippet, IconTimeline, IconTransfer, IconTrash, IconTrashOpen, IconTravelPlaces, IconTriangle, IconTruck, IconTwitter, IconTypography, IconTypostyles, IconUiPattern, IconUnderline, IconUnknown, IconUnknownSimple, IconUpload, IconUploadAlternative, IconUploadDoImages, IconUploadDontImages, IconUser, IconUserO, IconVideo, IconVideoAndText, IconVideoFullscreen, IconView, IconViewSlash, IconVimeo, IconVolumeOff, IconVolumeOn, IconWorkflow, IconZoom, InputLabel, ItemDragState, LegacyFlyoutFooter, LinkChooser, LoadingCircle, LoadingCircleSize, LoadingCircleStyle, MAX_STORED_ITEMS, MenuItem, MenuItemContent, MenuItemContentSize, MenuItemContentStyle, MenuItemStyle, MenuItemTextColorState, MultiInput, MultiInputLayout, MultiSelect, MultiSelectType, ON_SAVE_DELAY_IN_MS, OrderableList, QUERIES_STORAGE_KEY, RadioPill, RichTextEditor, SelectMenu, SelectionIndicatorIcon, SelectionMode, Slider, Stack, Switch, SwitchSize, Table, TableCell, TableCellType, TableColumnHeader, TableColumnHeaderType, TableHeaderRow, TableRow, Tag, TagType, Text, TextInput, TextInputType, Textarea, Tooltip, TooltipArrow, TooltipIcon, Tree, TreeNode, Trigger, TriggerSize, Validation, brightHeaderBackgroundColors, compose, debounce, generateRandomId, getColorDisplayValue, iconsMap, isColorLight, menuItemSizeClassMap, menuItemTextColorRecord, merge, sizeClasses, statusClasses, tagStyles, useClickOutside, useDebounce, useForwardedRef, useMemoizedId, validationClassMap)
```

File: package.json [hunk 1]
```diff
@@ -2,50 +2,36 @@
...
-        "./icons/*": {
-            "types": "./dist/src/foundation/Icon/*.d.ts",
-            "import": "./dist/src/foundation/Icon/*.js"
-        }
+        "./dist/*": "./dist/*"
     },
-    "types": "dist/index.d.ts",
     "scripts": {
-        "build": "npm run clean && npm run generate && cross-env NODE_ENV=production npm run build:library && npm run generate:barrelExport",
+        "build": "npm run clean && npm run generate:icons && npm run build:library",
```
[the remark this answers is not in the document]
Comment (SamuelAlev) [reply package.json:24]: Should be the default behavior, but I added it as part of `.env.production` to be safe

File: package.json [hunk 2]
```diff
@@ -2,50 +2,36 @@
     "name": "@frontify/arcade",
     "version": "2.1.0",
     "description": "Design system of Frontify",
-    "module": "dist/index.js",
+    "main": "dist/index.umd.js",
+    "unpkg": "dist/index.umd.js",
+    "jsdelivr": "dist/index.umd.js",
```
[the remark this answers is not in the document]
Comment (SamuelAlev) [reply package.json:7]: I just add it by default now so external devs who want to use it in `<script>` have the correct file by default now that we have 3 output format.

File: src/components/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+export * from "./MultiSelect";
+export * from "./OrderableList";
+export * from "./RadioPill";
+export * from "./RichTextEditor";
+export * from "./Slider";
+export * from "./Switch";
+export * from "./Table";
+export * from "./Tag";
+export * from "./TextInput";
+export * from "./Textarea";
```
[the remark this answers is not in the document]
Comment (SamuelAlev) [reply src/components/index.ts:35]: Done

File: vite.config.ts [hunk 4]
```diff
@@ -1,65 +1,54 @@
...
+            fileName: (format: string) => `index.${format}.js`,
+            formats: ["es", "umd", "cjs"],
+            name: "Arcade",
+        },
         sourcemap: true,
         minify: true,
         rollupOptions: {
-            input: [...componentsPath.map((element) => element.path), "src/styles.css"],
-            external: peerDependencies,
+            external: [...dependencies, ...peerDependencies, "react-color/lib/components/common"],
```
[the remark this answers is not in the document]
Comment (SamuelAlev) [reply vite.config.ts:42]: It was not marked as external because `'react-color' !== 'react-color/lib/components/common'`

**Review by jonasbuechel — CHANGES_REQUESTED** [review]
👍 I was able to build clarify frontend now, thx for the fix

ℹ️  We have to adjust tailwind config in Clarify once we update it. So it becomes a breaking change
Edit: I just updated the PR title


Changes in Clarify:
```
===================================================================
diff --git a/tailwind.config.js b/tailwind.config.js
--- a/tailwind.config.js	(revision 48675ff1d9fcee5cdd4e7a1d822981028ea96e92)
+++ b/tailwind.config.js	(date 1646064507419)
@@ -4,7 +4,7 @@
 
 module.exports = {
-    content: ['application/client/**/*.{js,ts,tsx}', 'node_modules/@frontify/arcade/dist/src/**/*.js'],
+    content: ['application/client/**/*.{js,ts,tsx}', 'node_modules/@frontify/arcade/dist/index.es.js'],
```

File: vite.config.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jonasbuechel) [reply vite.config.ts:42]: 👍

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonasbuechel) [reply package.json:24]: Fine, [doublechecked here](https://vitejs.dev/guide/env-and-mode.html#production-replacement) 👍

**Review by jonasbuechel — APPROVED** [review]
Successfully tested in standalone and integrated in Clarify 🚀

**Comment by SamuelAlev** [comment]
PR is done for clarify, I'll make a new release

**Pull request merged**